src/ui/control/geolocate_control.js

```diff
@@ -287,17 +288,17 @@ class GeolocateControl extends Evented {
     }
 
     _setupUI(supported: boolean) {
-        if (supported === false) {
-            warnOnce('Geolocation support is not available, the GeolocateControl will not be visible.');
-            return;
-        }
         this._container.addEventListener('contextmenu', (e: MouseEvent) => e.preventDefault());
         this._geolocateButton = DOM.create('button',
             `${className}-icon ${className}-geolocate`,
             this._container);
         this._geolocateButton.type = 'button';
         this._geolocateButton.title = 'Find my location';
         this._geolocateButton.setAttribute('aria-label', 'Find my location');
+        if (supported === false) {
+            warnOnce('Geolocation support is not available, the GeolocateControl will be disabled.');
```

Comment:
Nit: I know this wasn't introduced here, but it bugs me. Can we either make this two sentences or make it `so the GeolocateControl...`?